agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/RuntimeConfigurator.java

```diff
@@ -112,6 +133,31 @@ public void apply(RuntimeConfiguration runtimeConfig) {
       updateSampling(enabled, runtimeConfig.sampling, runtimeConfig.samplingPreview);
     }
 
+    // initialize Profiler
+    if (runtimeConfig.profilerEnabled && telemetryClient.getConnectionString() != null) {
```

Comment:
boolean in RuntimeConfigurator to avoid re-initializing profiler

similar for heartbeat below